### PR TITLE
Fix NaNs in grad(jax.nn.elu) for large positive inputs.

### PR DESCRIPTION
**Pull request by jheek**
The gradient of np.where is not robust to NaNs in the unselected branch.
A double where trick makes sure that the `expm1` will not NaN.
I have added a check_grads test that NaNs with the original implementation.

### CODE REVIEW

**Comment by mattjj** [comment]
See also [this comment](https://github.com/google/jax/issues/1052#issuecomment-514083352) on #1052 (and #347 and its "fix" #383 that I still plan to roll back) for a discussion.

**Review by mattjj — APPROVED** [review]
LGTM!
